hack/test.sh

```diff
@@ -560,6 +564,17 @@ deploy_workload_using_init_container() {
 
 # ------------------------------------------------------------------------------
 
+# Run Go unit tests before running more expensive tests.
+
+# Run Go unit tests
+echo "Running Go unit tests..."
+if ! go test ./...; then
+    echo "Go unit tests failed, exiting."
+    exit 1
+fi
```

Comment:
Runs unit tests before every integration test.

If unit tests are failing, there is no reason to run integration tests which are more expensive to run.